src/components/Tutorial/__tests__/Tutorial.ts

```diff
@@ -0,0 +1,296 @@
+import {
+  HOME_TOKEN,
+  TUTORIAL2_STEP_CHOOSE,
+  TUTORIAL2_STEP_CONTEXT1,
+  TUTORIAL2_STEP_CONTEXT1_PARENT,
+  TUTORIAL2_STEP_CONTEXT1_SUBTHOUGHT,
+  TUTORIAL2_STEP_CONTEXT2_PARENT,
+  TUTORIAL2_STEP_CONTEXT_VIEW_EXAMPLES,
+  TUTORIAL2_STEP_CONTEXT_VIEW_OPEN,
+  TUTORIAL2_STEP_CONTEXT_VIEW_SELECT,
+  TUTORIAL2_STEP_START,
+  TUTORIAL_STEP_AUTOEXPAND,
+  TUTORIAL_STEP_AUTOEXPAND_EXPAND,
+  TUTORIAL_STEP_FIRSTTHOUGHT,
+  TUTORIAL_STEP_FIRSTTHOUGHT_ENTER,
+  TUTORIAL_STEP_SECONDTHOUGHT,
+  TUTORIAL_STEP_SECONDTHOUGHT_ENTER,
+  TUTORIAL_STEP_START,
+  TUTORIAL_STEP_SUBTHOUGHT,
+  TUTORIAL_STEP_SUBTHOUGHT_ENTER,
+  TUTORIAL_STEP_SUCCESS,
+} from '../../../constants'
+import exportContext from '../../../selectors/exportContext'
+import store from '../../../stores/app'
+import createTestApp, { cleanupTestApp, cleanupTestEventHandlers } from '../../../test-helpers/createTestApp'
+
+// as per https://testing-library.com/docs/user-event/options/#advancetimers
+// we should avoid using { delay: null }, and use jest.advanceTimersByTime instead
+const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime })
+
+/** Get last created thought in the document, used mostly after `user.keyboard('{Enter}')`. */
+const lastThought = () => Array.from(document.querySelectorAll('[contenteditable="true"]')).at(-1)!
+
+/** Ensure we are at the given step of the tutorial. */
+const ensureStep = (step: number) => {
+  expect(store.getState().storageCache?.tutorialStep).toBe(step)
+}
+
+describe('Tutorial 1', async () => {
+  beforeEach(() => createTestApp({ tutorial: true }))
+  afterEach(cleanupTestEventHandlers)
+  afterAll(cleanupTestApp)
+  describe('step start', () => {
+    it('shows the welcome text', () => {
+      ensureStep(TUTORIAL_STEP_START)
+      expect(screen.getByText('Welcome to your personal thoughtspace.')).toBeInTheDocument()
+    })
+
+    it('shows the "Next" button, that leads to the first step', async () => {
+      await user.click(screen.getByText('Next'))
+      expect(screen.getByText(/First, let me show you how to create a new thought/)).toBeInTheDocument()
+    })
+  })
+
+  it('step first thought - how to create a thought ', async () => {
+    ensureStep(TUTORIAL_STEP_FIRSTTHOUGHT)
+    expect(screen.getByText('Hit the Enter key to create a new thought.')).toBeInTheDocument()
+    await user.keyboard('{Enter}')
+    expect(screen.getByText('You did it!')).toBeInTheDocument()
+
+    ensureStep(TUTORIAL_STEP_FIRSTTHOUGHT_ENTER)
+    expect(screen.getByText('Now type something. Anything will do.')).toBeInTheDocument()
+    await user.type(lastThought(), 'my first thought')
+    await user.keyboard('{Enter}')
+    await act(vi.runOnlyPendingTimersAsync)
+  })
+
+  it('step second thought - prompts to add another thought', async () => {
+    expect(screen.getByText('Well done!')).toBeInTheDocument()
+    expect(screen.getByText(/Try adding another thought/)).toBeInTheDocument()
+    ensureStep(TUTORIAL_STEP_SECONDTHOUGHT)
```

Comment:
Actions should generally be performed in the test that depends on them. So instead of typing and hitting enter in  one test and asserting the result in the next, move the typing and hitting enter to the next test where it is asserted.

As a general rule, a test should end with an `expect`. Anything after that should probably belong to the next test.